.github/workflows/pull-requests.yaml

```diff
@@ -1,89 +1,133 @@
+               "if": "matrix.host.cross_compile || matrix.host.platform_name == 'darwin_amd64'",
+               "name": "darwin_amd64: build${{ matrix.host.platform_name == 'darwin_amd64' && ' and test' || '' }}",
+               "run": "bazel ${{ matrix.host.bazel_startup_flags }} ${{ matrix.host.platform_name == 'darwin_amd64' && 'test --test_output=errors' || 'build' }} --platforms=@rules_go//go/toolchain:darwin_amd64 //..."
             },
             {
-               "name": "darwin_arm64: build and test",
-               "run": "bazel build --platforms=@rules_go//go/toolchain:darwin_arm64 //..."
+               "if": "matrix.host.cross_compile || matrix.host.platform_name == 'darwin_arm64'",
+               "name": "darwin_arm64: build${{ matrix.host.platform_name == 'darwin_arm64' && ' and test' || '' }}",
+               "run": "bazel ${{ matrix.host.bazel_startup_flags }} ${{ matrix.host.platform_name == 'darwin_arm64' && 'test --test_output=errors' || 'build' }} --platforms=@rules_go//go/toolchain:darwin_arm64 //..."
             },
             {
-               "name": "freebsd_amd64: build and test",
-               "run": "bazel build --platforms=@rules_go//go/toolchain:freebsd_amd64 //cmd/bb_copy //cmd/bb_replicator //cmd/bb_storage"
+               "if": "matrix.host.cross_compile || matrix.host.platform_name == 'freebsd_amd64'",
+               "name": "freebsd_amd64: build${{ matrix.host.platform_name == 'freebsd_amd64' && ' and test' || '' }}",
+               "run": "bazel ${{ matrix.host.bazel_startup_flags }} ${{ matrix.host.platform_name == 'freebsd_amd64' && 'test --test_output=errors' || 'build' }} --platforms=@rules_go//go/toolchain:freebsd_amd64 //..."
             },
             {
-               "name": "windows_amd64: build and test",
-               "run": "bazel build --platforms=@rules_go//go/toolchain:windows_amd64 //cmd/bb_copy //cmd/bb_replicator //cmd/bb_storage"
+               "if": "matrix.host.cross_compile || matrix.host.platform_name == 'windows_amd64'",
+               "name": "windows_amd64: build${{ matrix.host.platform_name == 'windows_amd64' && ' and test' || '' }}",
+               "run": "bazel ${{ matrix.host.bazel_startup_flags }} ${{ matrix.host.platform_name == 'windows_amd64' && 'test --test_output=errors' || 'build' }} --platforms=@rules_go//go/toolchain:windows_amd64 //..."
             }
-         ]
+         ],
+         "strategy": {
+            "matrix": {
+               "host": [
+                  {
+                     "bazel_os": "linux",
+                     "bazel_startup_flags": "",
+                     "cross_compile": true,
+                     "lint": true,
+                     "os": "ubuntu-latest",
+                     "platform_name": "linux_amd64",
+                     "upload": true
+                  },
+                  {
+                     "bazel_os": "windows",
+                     "bazel_startup_flags": "--output_base=D:/bazel_output",
```

Comment:
To prevent the `bazel` commands from getting messy, would it make sense to write this into `~/.bazelrc` instead?

Reply:
I can do but I'm not sure if it will improve it necessarily. We can't do:
```
startup:windows --output_base=
```
because that would force all Windows users to use that output base, even if no D drive existed.

What we could do is something like:
```
startup:gha-linux ...
startup:gha-windows ...
```
and then always add `--config`. But I'm not necessarily sure if that improves things.

What do you think?

Reply:
My thought was that the step that writes this line into .bazelrc could be marked:
```
"if": "matrix.host.platform_name == 'windows_amd64'",
```